### PR TITLE
Add Dataset integration tests - S3 Share requests

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix
- Tests

### Detail
- module `share_base`
- bugfix `delete_env` requires `env_object` not `envUri`
- TEMPORARY: hardcoded dataset_uri --> I wait for dataset module

### Relates
- #1376 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests_new/integration_tests/modules/share_base/fresh_deployment.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/fresh_deployment.py:1]: The naming does not match any of the other tests, we could call them something closer to the other modules `test_shares` and `test_shares_backwards_compatibility`

File: tests_new/integration_tests/modules/share_base/fresh_deployment.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/fresh_deployment.py:18]: We could also add assertions on the principal of the share and a new test for create_share_object for consumption roles that makes sure that the groupUri and the principal are correct

File: tests_new/integration_tests/modules/share_base/fresh_deployment.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/fresh_deployment.py:28]: Once the dataset PRs are merged we should add tests on getShareObject with the filters used to list shared items. Making sure it lists the tables, folders and bucket defining them explicitly instead of depending on a list of items

File: tests_new/integration_tests/modules/share_base/fresh_deployment.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/fresh_deployment.py:43]: My main concern is that we cannot test `test_reject_share` without running `test_submit_object_no_auto_approval` before. The tests are coupled, which in this case it might be alright, we are testing test_share_workflow succeed in parts. I would just clearly indicate that as a comment and group them all together (all tests on share1 together and all tests on share3 together to show the path). @petrkalos wdyt?

File: tests_new/integration_tests/modules/share_base/fresh_deployment.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/fresh_deployment.py:81]: Concern: how do we ensure there are items and this is not an empty list? As a next step after datasets PR, for the validation we would 1) s3 get object to the bucket and access point shares 2) run an athena query on the glue tables

File: tests_new/integration_tests/modules/share_base/input_types.py
Comment (dlpzx) [inline tests_new/integration_tests/modules/share_base/input_types.py:1]: Not sure having input_types is needed, we need a good reason to add more code

**Comment by dlpzx** [comment]
It would be good to have a testing plan on the different scenarios that we want to cover:
- Happy path approved
    - without Auto-approval
    - with Auto-approval
- Happy path revoked
- Unauthorized cases
- Error handling
For the different types of shareable objects:
- S3 Bucket
- S3 Prefix (folder)
- Glue table
For the different types of principals:
- group
- consumption role
For the different types of dataset:
- created
- imported with SSE-S3 encyrption
- imported with KMS encryption
- imported and Glue database imported

File: tests_new/integration_tests/conftest.py
Comment (dlpzx) [inline tests_new/integration_tests/conftest.py:17]: global conftests are meant to be used by the other modules. I am not sure we will want all the configs that we are creating in the other modules. Maybe we can have 2 conftest, one `global_conftest` (exportable to other modules) and a plain `conftest` (used internally in the shares tests). This is important because the share1, share3 states are very unpredictable

**Comment by dlpzx** [comment]
Issue to track PR: https://github.com/data-dot-all/dataall/issues/1409

File: tests_new/integration_tests/README.md
Comment (petrkalos) [inline tests_new/integration_tests/README.md:61]: I think the use of `aws_profiles` will not play very well with CodeBuild, instead I propose to use the existing infrastracture  (see `session_env1_aws_client`).

As discussed offline you need this account to test consumption roles. By using the integration test account you can solve it using the two following patterns
1. (simpler) add the already created (during env deployment) integration-test role directly as a consumption role. Current (CodeBuild/Local) account have already permissions to assume this role so you can use STS to assume it and then run S3 queries to make sure that the share was succesful.
2. (more complex) use the integration-test role to create new roles in the target account that you will register as consumption roles. Then proceed with assuming those roles and testing for S3 access.

**Review by petrkalos — CHANGES_REQUESTED** [review]

File: tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py
Comment (petrkalos) [inline tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py:41]: Instead of parametrizing the tests you can [parametrize a fixture](https://docs.pytest.org/en/latest/how-to/fixtures.html#parametrizing-fixtures) and then all tests that are using this fixture will run as many times as the parameters of the fixture.

For example in this case you can do something like...

```
@pytest.fixture(params=["Group", "ConsumptionRole"])
def principal1(request, group5, consumption_role_1):
    """
    :return: tuple with (principalUri, principalType)
    """
    if request.param is 'Group':
        yield (group5, request.param)
    else:
        yield (consumption_role_1.consumptionRoleUri, request.param)
```

File: tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py
Comment (petrkalos) [inline tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py:163]: Similar to my previous comment do you think it's possible to use parametrized fixtures here as well to avoid the `getfixturevalue`?

Comment (SofiaSazonova) [reply tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py]: implemented

File: tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py
Comment (SofiaSazonova) [reply tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py]: implemented

File: tests_new/integration_tests/README.md
Comment (SofiaSazonova) [reply tests_new/integration_tests/README.md]: I implemeted via AssumeRole

File: tests_new/integration_tests/aws_clients/iam.py
Comment (petrkalos) [inline tests_new/integration_tests/aws_clients/iam.py:32]: nit: logging

File: tests_new/integration_tests/aws_clients/iam.py
Comment (petrkalos) [inline tests_new/integration_tests/aws_clients/iam.py:48]: I am no IAM expert but do we need this? I think the first principal (line 46) will allow all roles from `account_id` to assume this role. Check [this](https://niebardzo.github.io/2021-08-23-root-principal-in-aws/#:~:text=IAM%20role%20assigned.-,Root%20Principal,-Who%20is%20the)

File: tests_new/integration_tests/aws_clients/sts.py
Comment (petrkalos) [inline tests_new/integration_tests/aws_clients/sts.py:7]: nit: shorthand...
```python
    def __init__(self, region, session = boto3.Session()):
       ...
```

File: tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py
Comment (petrkalos) [inline tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py:1]: this looks awesome :+1:

File: tests_new/integration_tests/aws_clients/iam.py
Comment (petrkalos) [inline tests_new/integration_tests/aws_clients/iam.py:43]: nit: I'd make this a dict and then do a `json.dumps`

**Review by petrkalos — CHANGES_REQUESTED** [review]

File: tests_new/integration_tests/README.md
Comment (SofiaSazonova) [reply tests_new/integration_tests/README.md:71]: 1. Speed up. 
2. Later we will need persistent shares as well

File: tests_new/integration_tests/aws_clients/iam.py
Comment (SofiaSazonova) [reply tests_new/integration_tests/aws_clients/iam.py:48]: No, it doesn't work. Assumed role is processed differently. I tried without it and got Access Denied, so I had to explicitly add this.

File: tests_new/integration_tests/README.md
Comment (petrkalos) [reply tests_new/integration_tests/README.md:71]: I think @dlpzx made a very good point in another PR about using persistent env/shares.
If we use them without forcing an update (and wait for it to complete) then we might not be testing the latest changes but if we do that then we might as well create a new env every time.

I think we should be able to use persistent envs with the argument of speed only for not very significant features AND obviously to test backwards compatibility (but for that we should still force update and wait).

File: tests_new/integration_tests/aws_clients/iam.py
Comment (SofiaSazonova) [reply tests_new/integration_tests/aws_clients/iam.py]: done

File: tests_new/integration_tests/aws_clients/sts.py
Comment (SofiaSazonova) [reply tests_new/integration_tests/aws_clients/sts.py]: done

File: tests_new/integration_tests/aws_clients/iam.py
Comment (SofiaSazonova) [reply tests_new/integration_tests/aws_clients/iam.py]: removed

**Review by petrkalos — APPROVED** [review]

**Pull request merged**
